contracts/src/interfaces/IStakeTable.sol

```diff
@@ -0,0 +1,130 @@
+        uint64 exitEpoch;
+        EdOnBN254.EdOnBN254Point schnorrVK;
+    }
+
+    // === Table State & Stats ===
+
+    /// @notice Get the commitment of the stake table used in current voting (i.e. snapshot at
+    /// the start of last epoch)
+    function votingStakeTableCommitment() external view returns (bytes32);
+    /// @notice Get the commitment of the stake table frozen for change (i.g. snapshot at the start
+    /// of the current epoch)
+    function frozenStakeTableCommitment() external view returns (bytes32);
+
+    /// @notice Get the total stakes of the registered keys in the latest stake table (Head).
+    /// @return The total stake for native token and restaked token respectively.
+    function totalStake() external view returns (uint256, uint256);
+    /// @notice Get the total number of stakers, uniquely identified by their `blsVK`
+    function totalKeys() external view returns (uint32);
+    /// @notice Get the total stake of the registered keys in the voting stake table
+    /// (LastEpochStart)
+    function totalVotingStake() external view returns (uint256);
+
+    /// @notice Look up the balance of `blsVK`
+    function lookup(BN254.G1Point calldata blsVK) external view returns (uint64);
+    /// @notice Look up the full `Node` state associated with `blsVK`
+    function fullLookup(BN254.G1Point calldata blsVK) external view returns (Node memory);
+
+    // === Queuing Stats ===
+
+    /// @notice Get the next available epoch for new registration
+    function nextRegisterationEpoch() external view returns (uint64);
+    /// @notice Get the number of pending registration requests in the waiting queue
+    function numPendingRegistrations() external view returns (uint64);
+    /// @notice Get the next available epoch for exit
+    function nextExitEpoch() external view returns (uint64);
+    /// @notice Get the number of pending exit requests in the waiting queue
+    function numPendingExit() external view returns (uint64);
+
+    // === Write APIs ===
+
+    /// @notice Register a validator in the stake table, transfer of tokens incurred!
+    ///
+    /// @param blsVK The BLS verification key
+    /// @param schnorrVK The Schnorr verification key (as the auxiliary info)
+    /// @param amount The amount to register
+    /// @param stakeType The type of staking (native or restaking)
+    /// @param blsSig The BLS signature that authenticates the `blsVK` field
+    /// @param maxWait The maximum epoch the sender is waiting to wait to be included (cannot be
+    /// smaller than the current epoch)
+    ///
+    /// @return success status
+    ///
+    /// @dev No validity check on `schnorrVK`, as it's assumed to be sender's responsibility,
+    /// the contract only treat it as auxiliary info submitted by `blsVK`.
+    function register(
+        BN254.G1Point calldata blsVK,
+        EdOnBN254.EdOnBN254Point calldata schnorrVK,
+        uint64 amount,
+        StakeType stakeType,
+        bytes calldata blsSig,
```

Comment:
Verifying a BLS signature inside the contract is expensive. As we do not have to perform BLS signatures aggregation anymore, why do we need this signature? If someone registers a BLS public key without knowing the corresponding private key, it will act as a byzantine node unable to sign a QC/proposal, is that a problem?

Reply:
> Verifying a BLS signature inside the contract is expensive.

this is not necessarily true. We can force every signature to be the same message, and precompute and hardcode the `HashToGroup(msg)` variable inside contract, then verifying the signature only takes 2 group validations and 2 pairings (via precompile)

> If someone registers a BLS public key without knowing the corresponding private key, it will act as a byzantine node unable to sign a QC/proposal, is that a problem?

Good point, I agree that we don't need to verify it anymore.
double checking with @jbearer

Reply:
Wait, we still do BLS aggregation. Not for the light client proofs but in consensus itself, to shrink the QCs

Reply:
Oh that is right, we should keep the check then.

Reply:
what? even if we do BLS aggregation, it has nothing to do with LightClient.sol as of now. 
We don't need to check them in contract, right? @jbearer

Reply:
We don't need to check the BLS signature in the contract but we need to be sure the registered participant knows the BLS private key to avoid key rogue attacks, unless this check is done somewhere else but it seems natural to do it here.

Reply:
Good point, in order to prevent Rogue key attack, we need to authenticate PoK on the secret key.

Reply:
added back in [65a87d0](https://github.com/EspressoSystems/espresso-sequencer/pull/748/commits/65a87d0b981f90a204a7ba399e1741653d9f1295)